reporter/datadog_reporter.go

```diff
@@ -559,12 +571,11 @@ func createPprofFunctionEntry(funcMap map[funcInfo]*pprofile.Function,
 	return function
 }
 
-//nolint:gocritic
-func addTraceLabels(labels map[string][]string, k traceAndMetaKey) {
+func addTraceLabels(labels map[string][]string, k *traceAndMetaKey,
```

Comment:
I changed the `traceAndMetaKey` to a pointer, highlighting in case there was any thought put into this earlier